Yosemite/YosemiteTests/Tools/Media/MediaImageExporterTests.swift

```diff
@@ -0,0 +1,84 @@
+import MobileCoreServices
+import XCTest
+@testable import Yosemite
+
+final class MediaImageExporterTests: XCTestCase {
+    func testExportingAnImageWithTypeHint() {
+        let mockData = UIImage(named: "image", in: Bundle(for: type(of: self)), compatibleWith: nil)!.pngData()
+        let filename = "test"
+        let typeHint = kUTTypeJPEG as String
+        let mockImageSourceWriter = MockupImageSourceWriter()
+        let exporter = MediaImageExporter(data: mockData!,
+                                          filename: filename,
+                                          typeHint: typeHint,
+                                          options: MediaImageExportOptions(maximumImageSize: nil,
+                                                                           imageCompressionQuality: 1.0,
+                                                                           stripsGeoLocationIfNeeded: true),
+                                          mediaDirectoryType: .temporary,
+                                          imageSourceWriter: mockImageSourceWriter)
+
+        do {
+            let expectedImageURL = try exporter.mediaFileManager
+                .createLocalMediaURL(filename: filename,
+                                     fileExtension: URL.fileExtensionForUTType(typeHint))
+            let expectation = self.expectation(description: "Export image data")
+            exporter.export { (uploadableMedia, error) in
+                XCTAssertEqual(mockImageSourceWriter.url, expectedImageURL)
+                expectation.fulfill()
+            }
+            waitForExpectations(timeout: Constants.expectationTimeout, handler: nil)
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testExportingAnImageWithoutTypeHint() {
```

Comment:
Is this the test that proves that `MediaImageExporter` supports `.heic` image files?

Reply:
it's actually the previous test case, `testExportingAnImageWithTypeHint` that supports HEIC images since we have to change the type to [one of the supported image files](https://en.support.wordpress.com/accepted-filetypes/) with JPEG as the default. The previous test case is trying to specify a type hint, so that the exporter attempts to export the image with the type hint. This case is when the type hint isn't specified, then the exporter just uses the type of the image file (which is PNG from the `pngData()` call in the following line. Lemme know if it's still not so clear what it is doing!

Reply:
Ah, thank you for the clarification. 🙇 

What do you think about renaming `testExportingAnImageWithTypeHint` to something that describes the business requirements?

```swift
func testItCanExportAnImageWithAnUnknownType() {
```

Reply:
thanks for the suggestion! I updated the function to `testExportingAnImageWithAnUnknownTypeHintUsesTheImageDataType` to be more descriptive and also added a comment in https://github.com/woocommerce/woocommerce-ios/pull/1776/commits/4eae48e2452a9da103970723136e9fdfdabae5c9